subsys/bluetooth/host/rfcomm.c

```diff
@@ -46,7 +46,7 @@
 #define RFCOMM_IDLE_TIMEOUT     K_SECONDS(2)
 
 #define DLC_RTX(_w) CONTAINER_OF(_w, struct bt_rfcomm_dlc, rtx_work)
-
+#define BR_CHAN(_ch) CONTAINER_OF(_ch, struct bt_l2cap_br_chan, chan)
```

Comment:
Looks like `BR_CHAN` should be moved to `l2cap_internal.h` instead of redefining it.

Reply:
Can we express the downcast macros as inline functions instead? Then we get the benefit of type checking the `_ch` argument.

Reply:
We could change them to functions, yes (doesn't need to be `inline` to work properly). Not sure if that should be done as part of this PR, but perhaps as a separate issue/PR

Reply:
Ok. Better to do it in a separate PR. We also have to see if we need to have two versions for const and non-const. (This may be a good use case for C11 generics if we need both).

@LingaoM Please move this line to `l2cap_internal.h`.